dotnet-client-libraries/ITextBasics/Tests/BigDocumentTest.cs

```diff
@@ -0,0 +1,18 @@
+﻿using ITextBasics;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Tests
+{
+    [TestClass]
+    public class BigDocumentTest
+    {
+        [TestMethod]
+        public void GivenCreateCommand_WhenRunning_ThenExpectCreatedFile()
+        {
+            var bigDocument = new BigDocument();
+            var tmpFileName = Path.GetTempFileName();
+            var runner = new PdfRunnerTest();
+            runner.DoTest(tmpFileName, bigDocument.Create);

```

Comment:
Files in TEMP folder do not have to be deleted. I chose to do so because I like to keep everything clean... to some extend 😉.

I would not use an external library for this, I would also not develop a new class with a disposable pattern just for that... No need to complicate things and program - clean code!

So I just solve it with a basic try - finally structure.

Reply:
While it is true, they do not have to be deleted, it is considered good practice to perform housekeeping. We can cause problems for other programs if there are too many files in $TEMP. In older versions of .NET there were issues related to creating too many that would cause software to fail. Related you can read [this answer on SO](https://stackoverflow.com/a/27708677).
Regarding an external library or an IDisposable, those were just suggestions of ways in which the problem could be solved. However you choose to address it is fine. I wrote a class years ago and published it in a few places (first to google code, then to CodeNet or whatever the MS one was called before they bought GitHub) and finally moved it all to GitHub.  It is a great utility class and I have needed it in lots of projects where I am dealing with large temp files and don't want to leak them into the TEMP folder. That is why I threw out that idea, thinking maybe you already have a utility class like that that you keep around. I like the IDisposable exactly for the purpose of clean code. The class is called TemporaryFile and does the very best it can to delete the file on Dispose 😀